### PR TITLE
Block storage review

### PR DESCRIPTION
**Pull request by iodine53**

### CODE REVIEW

File: SoftLayer/managers/block.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+            '4': '300'
+        }
+
+        prices = []
+        for item in package['items']:
+            for price in item['prices']:
+                for category in price['categories']:
+                    if price['locationGroupId'] == '' \
+                       and (category[
+                            'categoryCode'] == 'storage_service_enterprise'
```
Comment (tdurden82) [inline SoftLayer/managers/block.py]: pylint doesn't like this, but strangely doesn't object to the same thing on 237 *shrug*

File: SoftLayer/managers/block.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+                'snapshots.storageType.keyName',
+                'snapshots.snapshotCreationTimestamp',
+                'snapshots[hourlySchedule,dailySchedule,weeklySchedule]',
+            ]
+            kwargs['mask'] = "mask[%s]" % ','.join(items)
+        return self.block_svc.getObject(id=volume_id, **kwargs)
+
+    def order_block_volume(self, storage_type, location, size, os_type,
+                           iops=None, tier_level=None, **kwargs):
+
```
Comment (tdurden82) [inline SoftLayer/managers/block.py]: I guess I forgot the docstring on this one

File: SoftLayer/managers/block.py (hunk 1 above)
Comment (iodine53) [inline SoftLayer/managers/block.py]: The PEP8 tests which run via tox all but demanded that it be formatted thusly.  I'm not sure how better to handle it without breaking the 80-line width.  Refactor the _entire thing_?

File: SoftLayer/managers/block.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        base_type_name = 'SoftLayer_Container_Product_Order_Network_'
+        package = self._get_package(storage_type)
+        if package['name'] == 'Performance':
+            complexType = base_type_name + 'PerformanceStorage_Iscsi'
```
Comment (tdurden82) [inline SoftLayer/managers/block.py]: s/complexType/complex_type

File: SoftLayer/managers/block.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+            prices = self._get_item_prices_performance(package, size, iops)
+        if package['name'] == 'Endurance':
+            complexType = base_type_name + 'Storage_Enterprise'
```
Comment (tdurden82) [inline SoftLayer/managers/block.py]: s/complexType/complex_type

File: SoftLayer/managers/block.py (hunk 3 above)
Comment (iodine53) [inline SoftLayer/managers/block.py]: Accursed python.

**Pull request closed without merging**